src/Compilers/VisualBasic/Portable/Symbols/Tuples/TupleTypeSymbol.vb

```diff
@@ -0,0 +1,987 @@
+            Dim namesOfVirtualFields = ArrayBuilder(Of String).GetInstance(_elementTypes.Length)
+
+            If _elementNames.IsDefault Then
+                For i As Integer = 1 To _elementTypes.Length
+                    namesOfVirtualFields.Add(TupleMemberName(i))
+                Next
+            Else
+                namesOfVirtualFields.AddRange(_elementNames)
+            End If
+
+            Dim members = ArrayBuilder(Of Symbol).GetInstance(Math.Max(_elementTypes.Length, _underlyingType.OriginalDefinition.GetMembers().Length))
+
+            Dim currentUnderlying As NamedTypeSymbol = _underlyingType
+            Dim currentNestingLevel = 0
+
+            Dim currentFieldsForElements = ArrayBuilder(Of FieldSymbol).GetInstance(currentUnderlying.Arity)
+
+            ' Lookup field definitions that we are interested in
+            CollectTargetTupleFields(currentUnderlying, currentFieldsForElements)
+
+            Dim underlyingMembers As ImmutableArray(Of Symbol) = currentUnderlying.OriginalDefinition.GetMembers()
+
+            Do
+                For Each member In underlyingMembers
+                    Select Case member.Kind
+                        Case SymbolKind.Method
+                            If currentNestingLevel = 0 Then
+                                members.Add(New TupleMethodSymbol(Me, DirectCast(member, MethodSymbol).AsMember(currentUnderlying)))
+                            End If
+
+                        Case SymbolKind.Field
+                            Dim field = DirectCast(member, FieldSymbol)
+
+                            Dim tupleFieldIndex = currentFieldsForElements.IndexOf(field, ReferenceEqualityComparer.Instance)
+                            If tupleFieldIndex >= 0 Then
+                                ' This Is a tuple backing field
+                                Dim FieldSymbol = field.AsMember(currentUnderlying)
+
+                                If currentNestingLevel <> 0 Then
+                                    ' This is a matching field, but it is in the extension tuple
+                                    tupleFieldIndex += (RestPosition - 1) * currentNestingLevel
+
+
+                                    Dim defaultName As String = TupleMemberName(tupleFieldIndex + 1)
+                                    ' Add a field with default name if the given name Is different
+                                    If namesOfVirtualFields(tupleFieldIndex) <> defaultName Then
+                                        ' The name given doesn't match default name ItemTupleTypeSymbol.RestPosition, etc.
+                                        members.Add(New TupleRenamedElementFieldSymbol(Me, FieldSymbol, defaultName, -members.Count - 1, Nothing))
+                                    End If
+
+                                    ' Add a field with the given name
+                                    Dim Location = If(_elementLocations.IsDefault, Nothing, _elementLocations(tupleFieldIndex))
```

Comment:
done

---
In reply to: [71569677](https://github.com/dotnet/roslyn/pull/12596#discussion_r71569677) [](ancestors = 71569677)